tpot/tpot.py

```diff
@@ -496,6 +500,181 @@ def _random_forest(self, input_df, max_features):
         return self._train_model_and_predict(input_df, RandomForestClassifier, n_estimators=500,
                                              max_features=max_features, random_state=42, n_jobs=-1)
 
+    def _ada_boost(self, input_df, learning_rate):
+        """Fits an AdaBoost classifier
+
+        Parameters
+        ----------
+        input_df: pandas.DataFrame {n_samples, n_features+['class', 'group', 'guess']}
+            Input DataFrame for fitting the classifier
+        learning_rate: float
+            Learning rate shrinks the contribution of each classifier by learning_rate.
+
+        Returns
+        -------
+        input_df: pandas.DataFrame {n_samples, n_features+['guess', 'group', 'class', 'SyntheticFeature']}
+            Returns a modified input DataFrame with the guess column updated according to the classifier's predictions.
+            Also adds the classifiers's predictions as a 'SyntheticFeature' column.
+
+        """
+        return self._train_model_and_predict(input_df, AdaBoostClassifier,
```

Comment:
`learning_rate` should be capped at > 0.

Looking at the docs further, I also think that for the AdaBoostClassifier, we should allow `n_estimators` to be evolved as well, with a max of 500 estimators. The AdaBoostClassifier is one unique case where there is a tradeoff between `n_estimators` and `learning_rate`.